### ISSUE TITLE
Do not require --credentials-file with --validate-only

### ISSUE THREAD

**Issue by davepage-mcr** [pwaring/dns-sync-zone#46]
When running the script in `--validate-only` mode, it does not attempt to talk to Mythic Beasts' API. However, you are still required to pass a `--credentials-file` argument, even if it is nonsense:

```
$ python3 sync-zone.py --validate-only --zone example.com --zone-file /tmp/foo
sync-zone.py: error: the following arguments are required: --credentials-file

$ python3 sync-zone.py --validate-only --zone example.com --zone-file /tmp/foo --credentials-file /dev/null
```

The script should not require a `--credentials-file` argument if `--validate-only` is passed.

**Comment by pwaring** [comment]
Linking this to #42 as fixing that should also fix this issue (I need to investigate sub-parsers in `argparse` but haven't had chance to do so).

**Comment by pwaring** [comment]
This should be fixed in 3.0.0, please can you let me know if it works for you or causes any problems?

Your example above changes from:

```
python3 sync-zone.py --validate-only --zone example.com --zone-file /tmp/foo --credentials-file /dev/null
```

to:

```
python3 sync-zone.py validate --zone example.com --zone-file /tmp/foo
```
